Dockerfile

```diff
@@ -1,101 +1,66 @@
 #### Common Build Image Stage ####
 
-FROM ruby:2.7.1-alpine AS common-build-env
+FROM ruby:2.7.1-alpine AS common-build
 
-ARG APP_HOME=/app
-ARG BUILD_PACKAGES="build-base"
-ARG DEV_PACKAGES="postgresql-dev git nodejs yarn graphviz ttf-ubuntu-font-family"
-ARG RUBY_PACKAGES="tzdata"
+ARG BUILD_DEPS="git gcc libc-dev make nodejs yarn postgresql-dev graphviz-dev graphviz-doc build-base libxml2-dev libxslt-dev ttf-ubuntu-font-family"
 ARG bundleWithout=""
 
-ENV BUNDLER_VERSION="2.1.4" \
-    BUNDLE_WITHOUT=${bundleWithout} \
-    WKHTMLTOPDF_GEM=wkhtmltopdf-binary-edge-alpine
-
-RUN apk update && \
-    apk upgrade && \
-    apk add --update --no-cache $BUILD_PACKAGES $DEV_PACKAGES $RUBY_PACKAGES && \
-    gem update --system && \
-    find / -wholename '*default/bundler-*.gemspec' -delete && \
-    rm /usr/local/bin/bundle && \
-    rm /usr/local/bin/bundler && \
-    gem install bundler -v $BUNDLER_VERSION
-
-WORKDIR $APP_HOME
-
-COPY Gemfile Gemfile.lock package.json yarn.lock ./
-
-RUN bundle install && \
-    yarn install
-
-
-#### Dev build builds on common build env image ####
-
-FROM common-build-env AS dev-build
-ARG APP_HOME=/app
-ENV RAILS_ENV=development
-
-WORKDIR $APP_HOME
-
-COPY . .
-
-CMD bundle exec rails db:migrate && bundle exec rails server -b 0.0.0.0
-
-
-#### Prod minify stage builds the app on the common build env image ####
-
-FROM common-build-env AS prod-minify
-
-ARG APP_HOME=/app
-
 # These variables are required for running Rails processes like assets:precompile
-ENV RAILS_ENV=production \
+ENV BUNDLE_WITHOUT=${bundleWithout} \
+    WKHTMLTOPDF_GEM=wkhtmltopdf-binary-edge-alpine \
+    RAILS_ENV=test \
     GOVUK_NOTIFY_API_KEY=TestKey \
-    AUTHORISED_HOSTS=dummy.build.domain \
+    AUTHORISED_HOSTS=127.0.0.1 \
     SECRET_KEY_BASE=TestKey \
     GOVUK_NOTIFY_CALLBACK_API_KEY=TestKey
 
-WORKDIR $APP_HOME
-
-RUN rm -rf /usr/local/bundle/cache/*.gem && \
-    find /usr/local/bundle/gems -name "*.c" -delete && \
-    find /usr/local/bundle/gems -name "*.h" -delete && \
-    find /usr/local/bundle/gems -name "*.o" -delete
+WORKDIR /app
 
 COPY . .
 
-RUN bundle exec rake assets:precompile
-
-RUN rm -rf tmp/cache app/assets lib/assets vendor/assets node_modules
-
-
-#### Production image builds from scratch and copies the build app components from the prod minify stage ####
+RUN apk update && \
+    apk upgrade && \
+    apk add --update --no-cache tzdata libpq libxml2 libxslt graphviz  && \
+    apk add --update --no-cache --virtual .gem-installdeps $BUILD_DEPS && \
+    gem update --system && \
+    find / -wholename '*default/bundler-*.gemspec' -delete && \
+    rm -rf /usr/local/bin/bundle && \
+    gem install bundler -v 2.1.4 && \
+    bundle install --no-binstubs --jobs=3 --retry=5 && \
+    yarn install --check-files && \
+    rm -rf yarn.lock && \
+    bundle exec rake assets:precompile && \
+    apk del .gem-installdeps && \
+    rm -rf tmp log app/assets lib/assets vendor/assets node_modules && \
```

Comment:
None of these files except `node_modules` exist in our app. We could probably remove `app/frontend` but it's only 160K.

Reply:
the tmp and lib folders where created in my local dev env during the build

Reply:
Sorry, yes — I missed those!

Reply:
removed `app/assets`, `lib/assets`, `vendor/assets` from the remove list and added `app/frontend` to be deleted